File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_map<int, int> freq;
        int max = 0;

        for (auto i : nums) {
            freq[i]++;
        }

        for (auto i : nums) {
            vector<int> res;
            if (!freq[i-1]) {
                res.push_back(i);
                int n = 1;
                while (freq[i+n]) {
                    res.push_back(i+n);
                    n++;
                }
                max = max > res.size() ? max : res.size();
            }
        }
        return max;
    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-3.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        f = {}
        res = []

        for i in nums:
            if i in f:
                f[i] += 1
            else:
                f[i] = 1
        
        for i in range(k):
            maxNum = nums[0]
            for j in f:
                if f[j] > f[maxNum]:
                    maxNum = j
            res.append(maxNum)
            f[maxNum] = 0
        
        return res
